src/bot.py

```diff
@@ -192,15 +196,50 @@ def is_last_run(self):
                 found_unfinished_run = True
                 break
         return not found_unfinished_run
+    
+    def _rebuild_as_asset_to_trigger(trigger_to_assets: dict):
+        result = {}
+        for key in trigger_to_assets.keys():
+            for asset in trigger_to_assets[key]:
+                result[asset] = key
+        return result
+
+    def on_init(self):
+        keyboard.release(self._config.char["stand_still"])
+        transition_to_screens = Bot._rebuild_as_asset_to_trigger({
+            "select_character": [
+                "MAIN_MENU_TOP_LEFT","MAIN_MENU_TOP_LEFT_DARK"
+            ],
+            "start_from_town": [
+                 "A5_TOWN_0", "A5_TOWN_1",
+                 "A4_TOWN_4", "A4_TOWN_5",
+                 "A3_TOWN_0", "A3_TOWN_1",
+                 "A2_TOWN_0", "A2_TOWN_1", 
+                 "A2_TOWN_10","A1_TOWN_1", "A1_TOWN_3"],
+        })
+        match = self._template_finder.search_and_wait(list(transition_to_screens.keys()), best_match=True)
+        self.trigger_or_stop(transition_to_screens[match.name])
+    
+    def on_select_character(self):
+        if self._config.general['restart_d2r_when_stuck']:
+            # Make sure the correct char is selected
+            if self._char_selector.has_char_template_saved():
+                Logger.info("Selecting original char")
+                self._char_selector.select_char()
+            else:
+                Logger.info("Saving top-most char as template")
+                self._char_selector.save_char_template()
+        self.trigger_or_stop("create_game")
 
     def on_create_game(self):
-        keyboard.release(self._config.char["stand_still"])
         # Start a game from hero selection
-        self._game_stats.log_start_game()
         self._template_finder.search_and_wait(["MAIN_MENU_TOP_LEFT","MAIN_MENU_TOP_LEFT_DARK"], roi=self._config.ui_roi["main_menu_top_left"])
         if not self._ui_manager.start_game(): return
-        self._curr_loc = self._town_manager.wait_for_town_spawn()
+        self.trigger_or_stop("start_from_town")
 
+    def on_town_init(self):
```

Comment:
Rename to `on_start_from_town` please to keep naming convention constant. Or rename the state from start_from_town to town_init.